envoy/api/v2/endpoint/endpoint.proto

```diff
@@ -18,7 +18,15 @@ option (gogoproto.equal_all) = true;
 
 // Upstream host identifier.
 message Endpoint {
+  // The upstream host address.
   core.Address address = 1;
+
+  // [#not-implemented-hide:] Optional alternative health check port.
+  //
+  // By default the health check address port of an upstream host is the same as the host's serving
```

Comment:
Presumably we're using `uint32` rather than `google.protobuf.UInt32Value` because we can't possibly have a valid health check port of 0, since that doesn't make sense in TCP.